### PR TITLE
Combining steps to run kapp test in release.yml

### PR DESCRIPTION
**Pull request by rohitagg2020**
Combining steps to run kapp test in release.yml in a single step.

Signed-off-by: rohitagg2020 <rohit.aggarwal2020@gmail.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -24,8 +24,8 @@ jobs:
           go-version: 1.19.5
 
       - name: Run GoReleaser
-        # GoReleaser v4.2.0
-        uses: goreleaser/goreleaser-action@f82d6c1c344bcacabba2c841718984797f664a6b
+        # GoReleaser v2.5.0
+        uses: goreleaser/goreleaser-action@5e15885530fb01d81d1f24e8a6f54ebbd0fed7eb
```
Comment (praveenrewar) [inline .github/workflows/release.yml]: Should we have a separate commit for this (separation of concern and also to cherry pick)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -43,21 +43,20 @@ jobs:
           # Ensure that there is no existing kapp installed
           rm -f /tmp/bin/kapp
 
-      - name: Create kapp binary
+      - name: Run Test cases
```
Comment (praveenrewar) [inline .github/workflows/release.yml]: Should we also move this step to the end? The tests can fail due to various reasons and sometimes we might wanna go ahead with the release even if the tests have failed, so if we move this step to the end then we will know if the checksums are matching. But even if the tests fail and the remaining steps are still executed, then I guess it's fine to have it here itself.

File: .github/workflows/release.yml (hunk 1 above)
Comment (rohitagg2020) [reply .github/workflows/release.yml:28]: Do we want to squash these commits later?

**Comment by rohitagg2020** [comment]
Raised another [PR](https://github.com/carvel-dev/kapp/pull/713).

**Pull request closed without merging**